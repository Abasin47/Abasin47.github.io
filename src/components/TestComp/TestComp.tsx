import * as React from 'react';
import styled from 'styled-components';

import { colors } from '../../assets/styles';

interface ITestCompProps {
    className?: string;
}

const P = styled.p`
    color: ${colors.baseFontColor};
`;

const TestComp: React.FC<ITestCompProps> = () => {
    return <P>What a happy day</P>;
};

export { TestComp };
