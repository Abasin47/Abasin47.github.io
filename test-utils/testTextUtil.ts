export const testInitalMessages = {
    'test.text': 'test text',
    'test.tooltip': 'test tooltip',
    'test.dummy': 'test dummy',
    'test.key': 'test key'
};
