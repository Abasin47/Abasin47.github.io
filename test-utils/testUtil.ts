/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import { render, queries, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { queries: { ...queries }, ...options });

export * from '@testing-library/react';
export { customRender as render };
