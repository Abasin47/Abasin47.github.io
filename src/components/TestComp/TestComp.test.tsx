import * as React from 'react';

import { render } from '../../../test-utils/testUtil';
import { TestComp } from './TestComp';

describe(TestComp, () => {
    test('snapshot', () => {
        const wrapper = render(<TestComp />);
        expect(wrapper.asFragment()).toMatchSnapshot();
    });
});
