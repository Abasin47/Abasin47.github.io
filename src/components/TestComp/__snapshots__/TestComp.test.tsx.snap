// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComp snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: #ffffff;
}

<p
    class="c0"
  >
    What a happy day
  </p>
</DocumentFragment>
`;
